org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/internal/DefaultModelPersistence.java

```diff
@@ -54,32 +52,33 @@
 public final class DefaultModelPersistence implements ModelPersistence, EventListener {
 
     // Cache holding (project name - persistent model property) pairs.
```

Comment:
You could also have 

```
PersistentModelBuilder {
  PersistentModel getPrevious()
}
```

Then you could check `builder.previous.present?` instead of doing null checks.